python/ray/experimental/slib/streaming.py

```diff
@@ -0,0 +1,339 @@
+import sys
+import uuid
+import networkx as nx
+
+from ray.experimental.slib.operator import *
+from ray.experimental.slib.actor import *
+from ray.experimental.slib.batched_queue import BatchedQueue
+
+# Generates UUIDs
+def generate_uuid():
+    return uuid.uuid4()
+
+# Environment configuration
+class Config(object):
+    def __init__(self,parallelism=1):
+        self.queue_config = QueueConfig()
+        self.parallelism = parallelism
+        # ...
+
+# The execution environment for a streaming job
+class Environment(object):
+    def __init__(self, config=Config()):
+        self.logical_topo = nx.DiGraph()    # DAG
+        self.physical_topo = nx.DiGraph()   # DAG
+        self.operators = {}                 # operator id --> operator object
+        self.config = config                # Environment's configuration
+        self.topo_cleaned = False           # True if logical dataflow graph is ready for execution, false otherwise
+        self.actor_handles = []             # Handles to all actors in the physical dataflow
+
+    # Generates all required data channels between an operator and its downstream operators
+    def __generate_channels(self, operator):
```

Comment:
Add docstrings for these methods using the same format we use for other docstrings